src/domain/chains/entities/__tests__/chain.builder.ts

```diff
@@ -10,7 +10,7 @@ import { rpcUriBuilder } from './rpc-uri.builder';
 
 export function chainBuilder(): IBuilder<Chain> {
   return Builder.new<Chain>()
-    .with('chainId', faker.random.numeric())
+    .with('chainId', faker.random.numeric(6))
```

Comment:
This change is motivated because I was getting random errors (flaky tests) due to the cache contents being different from the expected ones. This was caused because `faker.random.numeric` only produces one digit by default, so sometimes two sequential calls to `chainBuilder()` produce the same digit, and the cache direction gets repeated. So only one chain was being cached (the same cache identifier was produced twice).

To avoid this situation, the simplest fix is increasing the entropy and making `faker.random.numeric` to produce 6 digits. Collisions are very rare that way, but we could also think on producing an uuid here, even if this would make our `chainBuilder` differ from the production behaviour, where chainIds are integers.